lib/V8/processMonitor.cpp

```diff
@@ -0,0 +1,170 @@
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <algorithm>
+#include <chrono>
+#include <memory>
+#include <thread>
+#include <type_traits>
+
+#include "Basics/process-utils.h"
+#include "Basics/signals.h"
+#include "Basics/system-functions.h"
+
+#if defined(TRI_HAVE_MACOS_MEM_STATS)
+#include <sys/sysctl.h>
+#endif
+
+#ifdef TRI_HAVE_SYS_PRCTL_H
+#include <sys/prctl.h>
+#endif
+
+#ifdef TRI_HAVE_SIGNAL_H
+#include <signal.h>
+#endif
+
+#ifdef TRI_HAVE_SYS_WAIT_H
+#include <sys/wait.h>
+#endif
+
+#ifdef TRI_HAVE_MACH
+#include <mach/mach_host.h>
+#include <mach/mach_port.h>
+#include <mach/mach_traps.h>
+#include <mach/task.h>
+#include <mach/thread_act.h>
+#include <mach/vm_map.h>
+#endif
+
+#ifdef _WIN32
+#include <Psapi.h>
+#include <TlHelp32.h>
+#include "Basics/socket-utils.h"
+#include "Basics/win-utils.h"
+#endif
+#include <fcntl.h>
+
+#ifdef TRI_HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#include "Basics/Mutex.h"
+#include "Basics/MutexLocker.h"
+#include "Basics/DeadlockDetector.h"
+#include "Basics/Thread.h"
+#include "V8/processMonitor.h"
+#include "V8/v8-deadline.h"
+
+using namespace arangodb;
+
+namespace {
+
+/// @brief lock for protected access to vector ExternalProcesses
+static arangodb::Mutex ExitedExternalProcessesLock;
+
+}  // namespace
+
+namespace arangodb {
+
+/// @brief if monitored & exited we keep them here:
+std::map<TRI_pid_t, ExternalProcessStatus> ExitedExternalProcessStatus;
+
+std::vector<ExternalId> monitoredProcesses;
+
+void addMonitorPID(ExternalId& pid) {
+  MUTEX_LOCKER(mutexLocker, ExitedExternalProcessesLock);
+  monitoredProcesses.push_back(pid);
+}
+
+void removeMonitorPID(ExternalId const& pid) {
+  MUTEX_LOCKER(mutexLocker, ExitedExternalProcessesLock);
+  for (auto it = monitoredProcesses.begin(); it != monitoredProcesses.end();
+       ++it) {
+    if (it->_pid == pid._pid) {
+      monitoredProcesses.erase(it);
+      break;
+    }
+  }
+}
+
+std::optional<ExternalProcessStatus> getHistoricStatus(TRI_pid_t pid) {
+  MUTEX_LOCKER(mutexLocker, ExitedExternalProcessesLock);
+  auto it = ExitedExternalProcessStatus.find(pid);
+  if (it == ExitedExternalProcessStatus.end()) {
+    return std::nullopt;
+  }
+  return std::optional<ExternalProcessStatus>{it->second};
+}
+
+class ProcessMonitorThread : public arangodb::Thread {
+ public:
+  ProcessMonitorThread(application_features::ApplicationServer& server)
+      : Thread(server, "ProcessMonitorThread") {}
```

Comment:
There is only limited space for thread names. Normally suffixes such as "Thread" are omitted from the names, as they are not adding more insight or value.
```suggestion
      : Thread(server, "ProcessMonitor") {}
```